great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -1072,7 +1163,8 @@ def close(self) -> None:
         More background can be found here: https://github.com/great-expectations/great_expectations/pull/3104/
         """
         if self._engine_backup:
-            self.engine.close()
+            if self._connection:
```

Comment:
I think we can get rid of this method - in a followup PR since this PR is already quite large (see method docstring).

Reply:
Yes I agree. Let's get rid of it soon.